src/components/modules/caret.ts

```diff
@@ -548,6 +558,117 @@ export default class Caret extends Module {
     selection.addRange(newRange);
   }
 
+  /**
+   * Detect an next line position from the caret position
+   */
+  private detectNextLinePosition(): Position | false {
+    const { BlockManager, Tools } = this.Editor;
+
+    const nextInput = BlockManager.currentBlock.nextInput;
+    let nextBlock = BlockManager.nextContentfulBlock;
+
+    if (!nextBlock && !nextInput) {
+      /**
+       * If there is a last block and it is default, do nothing
+       * This code allows to exit from the last non-initial tool:
+       * https://github.com/codex-team/editor.js/issues/1103
+       */
+      if (BlockManager.currentBlock.tool.isDefault) {
```

Comment:
in the previous version, there also was a [statement](https://github.com/codex-team/editor.js/pull/1526/files#diff-e5b85a44b7feee1d984858fa440a74cf3373fdf8d12c4e229051361b3ce97c04L412) checking isAtEnd position. Is this case still covered?

Reply:
Seems to be not relevant anymore as we just need to check if there is next line or not